atomic_defi_design/qml/Components/DexModal.qml

```diff
@@ -1,33 +1,59 @@
-import QtQuick 2.15
-import QtQuick.Controls 2.15
-import "../Constants"
-
-Popup {
-    id: root
-    anchors.centerIn: Overlay.overlay
-    modal: true
-    focus: true
-    closePolicy: Popup.CloseOnEscape | Popup.CloseOnPressOutside
-
-    Overlay.modal: Rectangle {
-        color: "#AA000000"
-    }
-
-    // Fade in animation
-    onVisibleChanged: {
-        if(visible) {
-            opacity = 0
-            fade_animation.start()
-        }
-    }
-
-    NumberAnimation {
-        id: fade_animation
-        target: root
-        property: "opacity"
-        duration: Style.animationDuration
-        to: 1
-    }
-
-    background: FloatingBackground { }
-}
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+Popup {
+	id: control
+	parent: Overlay.overlay
+	property alias title: _headerBackground.text
+	property alias headerBackground: _headerBackground
+	property alias backgroundColor: _backgroundColor.color
+	property alias currentIndex: _layoutPopup.currentIndex
+	property alias header: _header.contentItem
+	property alias footer: _footer.contentItem
+	modal: true
+	padding: 0
+	Overlay.modeless: DexRectangle {
+        color: theme.dexBoxBackgroundColor
+        opacity: .3
+    }
+	background: ClipRRect {
+		radius: 8
+		DexRectangle { 
+			id: _backgroundColor
+			anchors.fill: parent
+			border.width: 2
+			radius: parent.radius
+			color: theme.dexBoxBackgroundColor
+			Container {
+				id: _header 
+				width: parent.width
+				height: 60
+				contentItem: DexModalHeader {
+					id: _headerBackground
+				}
+			}
+			HorizontalLine {
+				anchors.top: _footer.top
+				width: _headerBackground.width 
+				opacity: .7
+			}
+			Container {
+				id: _footer
+				anchors.bottom: parent.bottom
+				height: try { contentItem.height } catch(e){ 0 }
```

Comment:
There is no purpose of doing a `try catch` block if you do not need any logic in your `catch`.